dll/win32/shell32/shellmenu/CStartMenu.cpp

```diff
@@ -238,19 +265,65 @@ class CShellMenuCallback :
+            PathAppendW(pszPath, szTitle);
+            StringCchCatW(pszPath, cchPath, L".lnk");
+            return PathFileExistsW(pszPath);
+        }
+        return FALSE;
+    }
+
     HRESULT OnExec(LPSMDATA psmd)
     {
         // HACK: Because our ShellExecute can't handle CLSID components in paths, we can't launch the paths using the "open" verb.
         // FIXME: Change this back to using the path as the filename and the "open" verb, once ShellExecute can handle CLSID path components.
 
-        if (psmd->uId == IDM_CONTROLPANEL)
-            ShellExecuteW(NULL, NULL, L"explorer.exe", L"::{20D04FE0-3AEA-1069-A2D8-08002B30309D}\\::{21EC2020-3AEA-1069-A2DD-08002B30309D}", NULL, SW_SHOWNORMAL);
-        else if (psmd->uId == IDM_NETWORKCONNECTIONS)
-            ShellExecuteW(NULL, NULL, L"explorer.exe", L"::{20D04FE0-3AEA-1069-A2D8-08002B30309D}\\::{21EC2020-3AEA-1069-A2DD-08002B30309D}\\::{7007ACC7-3202-11D1-AAD2-00805FC1270E}", NULL, SW_SHOWNORMAL);
-        else if (psmd->uId == IDM_PRINTERSANDFAXES)
-            ShellExecuteW(NULL, NULL, L"explorer.exe", L"::{20D04FE0-3AEA-1069-A2D8-08002B30309D}\\::{21EC2020-3AEA-1069-A2DD-08002B30309D}\\::{2227A280-3AEA-1069-A2DE-08002B30309D}", NULL, SW_SHOWNORMAL);
-        else
-            PostMessageW(m_hwndTray, WM_COMMAND, psmd->uId, 0);
+        WCHAR szPath[MAX_PATH];
+        switch (psmd->uId)
+        {
+            case IDM_CONTROLPANEL:
+                ShellExecuteW(NULL, NULL, L"explorer.exe", L"::{20D04FE0-3AEA-1069-A2D8-08002B30309D}\\::{21EC2020-3AEA-1069-A2DD-08002B30309D}", NULL, SW_SHOWNORMAL);
+                break;
+            case IDM_NETWORKCONNECTIONS:
+                ShellExecuteW(NULL, NULL, L"explorer.exe", L"::{20D04FE0-3AEA-1069-A2D8-08002B30309D}\\::{21EC2020-3AEA-1069-A2DD-08002B30309D}\\::{7007ACC7-3202-11D1-AAD2-00805FC1270E}", NULL, SW_SHOWNORMAL);
+                break;
+            case IDM_PRINTERSANDFAXES:
+                ShellExecuteW(NULL, NULL, L"explorer.exe", L"::{20D04FE0-3AEA-1069-A2D8-08002B30309D}\\::{21EC2020-3AEA-1069-A2DD-08002B30309D}\\::{2227A280-3AEA-1069-A2DE-08002B30309D}", NULL, SW_SHOWNORMAL);
+                break;
+            case IDM_MYDOCUMENTS:
+                SHGetSpecialFolderPathW(NULL, szPath, CSIDL_PERSONAL, FALSE);
+                ShellExecuteW(NULL, NULL, szPath, NULL, NULL, SW_SHOWNORMAL);
+                break;
+            case IDM_MYPICTURES:
+                SHGetSpecialFolderPathW(NULL, szPath, CSIDL_MYPICTURES, FALSE);
+                ShellExecuteW(NULL, NULL, szPath, NULL, NULL, SW_SHOWNORMAL);
+                break;
+            case IDM_USERPROFILE:
+                SHGetSpecialFolderPathW(NULL, szPath, CSIDL_PROFILE, FALSE);
+                ShellExecuteW(NULL, NULL, szPath, NULL, NULL, SW_SHOWNORMAL);
+                break;
+            default:
+            {
+                if (IDM_RECENTMIN <= psmd->uId && psmd->uId <= IDM_RECENTMAX)
```

Comment:
Leave just PostMessageW in the default clause

Reply:
OK.